data/crac-io/crac-io-json/src/main/java/com/farao_community/farao/data/crac_io_json/deserializers/CracDeserializer.java

```diff
@@ -0,0 +1,102 @@
+ */
+public class CracDeserializer extends JsonDeserializer<Crac> {
+
+    private CracFactory cracFactory;
+
+    private CracDeserializer() {
+    }
+
+    public CracDeserializer(CracFactory cracFactory) {
+        this.cracFactory = cracFactory;
+    }
+
+    @Override
+    public Crac deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        jsonParser.nextToken();
+        while (!jsonParser.getCurrentName().equals(ID)) {
+            if (jsonParser.nextToken() == JsonToken.END_OBJECT) {
+                throw new FaraoException(String.format("The JSON Crac must contain an %s field", ID));
+            }
+            jsonParser.nextToken();
+        }
+        String id = jsonParser.nextTextValue();
+        jsonParser.nextToken();
+        if (!jsonParser.getCurrentName().equals(NAME)) {
+            throw new FaraoException(String.format("The JSON Crac must contain a %s field after the %s field", NAME, ID));
+        }
+        String name = jsonParser.nextTextValue();
+
+        Crac crac = cracFactory.create(id, name);
+
+        Map<String, String> deserializedNetworkElementsNamesPerId = null;
+        // deserialize the following lines of the Crac
+        while (jsonParser.nextToken() != JsonToken.END_OBJECT) {
+            switch (jsonParser.getCurrentName()) {
+                case NETWORK_ELEMENTS_NAME_PER_ID:
+                    jsonParser.nextToken();
+                    deserializedNetworkElementsNamesPerId = jsonParser.readValueAs(HashMap.class);
+                    break;
+
+                case CONTINGENCIES:
+                    jsonParser.nextToken();
+                    ContingencyArrayDeserializer.deserialize(jsonParser, crac, deserializedNetworkElementsNamesPerId);
+                    break;
+
+                case FLOW_CNECS:
+                    jsonParser.nextToken();
+                    FlowCnecArrayDeserializer.deserialize(jsonParser, deserializationContext, crac, deserializedNetworkElementsNamesPerId);
+                    break;
+
+                case PST_RANGE_ACTIONS:
+                    jsonParser.nextToken();
+                    PstRangeActionArrayDeserializer.deserialize(jsonParser, deserializationContext, crac, deserializedNetworkElementsNamesPerId);
+                    break;
+
+                case NETWORK_ACTIONS:
+                    jsonParser.nextToken();
+                    NetworkActionArrayDeserializer.deserialize(jsonParser, deserializationContext, crac, deserializedNetworkElementsNamesPerId);
+                    break;
+
+                case EXTENSIONS:
+                    jsonParser.nextToken();
+                    List<Extension<Crac>> extensions = JsonUtil.readExtensions(jsonParser, deserializationContext, ExtensionsHandler.getExtensionsSerializers());
+                    ExtensionsHandler.getExtensionsSerializers().addExtensions(crac, extensions);
+                    break;
```

Comment:
I think this can be removed as we don't use CracResultExtension anymore

Reply:
We kept the extensions for now as they are still used in the flowbased-computation package